src/GeoIP.php

```diff
@@ -98,13 +104,20 @@ public function __construct(array $config, CacheManager $cache)
         $this->cache->setPrefix((string) $this->config('cache_prefix'));
 
         // Set custom default location
-        $this->default_location = array_merge(
-            $this->default_location,
-            $this->config('default_location', [])
-        );
+        $this->default_location = is_callable(self::$defaultLocationResolver)
+            ? call_user_func(self::$defaultLocationResolver)->toArray()
```

Comment:
Hey @alies-dev,

Does this mean that we now have two places to change this value?
Also, this discards one of the usecases that we have, which is changing the country for local testing & in tests.

Reply:
@mithredate 
yes, this is a public repo, I implemented this resolver and will remove other options on next major version of the package

Reply:
>Also, this discards one of the usecases that we have, which is changing the country for local testing & in tests.

I'm not sure it exactly discards, but changes a way on how to fake/set the response, right?

Reply:
What I mean is that this code is being committed to git. Currently, we use the env to change the default location for local development. This new snippet doesn't allow that, unless you ignore the file (unstage it). This is error-prone and not really a configuration to enhance DX IMO.

Reply:
Please give me some time to think about this. I also like the new approach better, but it doesn't help since we might accidentally commit things and then we need to roll-them back, etc.